src/utils/logging.cc

```diff
@@ -80,8 +124,23 @@ StderrLogMessage::StderrLogMessage(const char* file, int line)
     : log_(file, line) {}
 
 StderrLogMessage::~StderrLogMessage() {
-  std::cerr << str() << std::endl;
-  log_ << str();
+  std::string s = str();
+  std::string s2 = "";
+  size_t len;
```

Comment:
This should be factored out to a separate function.
Also this feels a bit wastful to do for every single log message (especially given that lots of string copying is going on, and we don't yet use std::string_view as we don't use C++17), but objectively there's no difference.